cloud-dataplane/overlays/update-dataplane-server-config.yaml

```
@@ -0,0 +1,20 @@
+actions:
+  # Update only the server URL template
+  - target: "$.servers[0].url"
+    update: "https://{dataplane_api_subdomain}.cloud.redpanda.com"
+  # Update only the variables section
+  - target: "$.servers[0].variables"
+    update:
+      dataplane_api_subdomain:
+        default: "{dataplane.api.subdomain}"
+        description: |-
+          From the `dataplane_api.url` value in the Control Plane API response, extract the subdomain (the part between `https://` and `.cloud.redpanda.com`). Enter this value in the Data Plane API URL field.
+          
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)